File: examples/app-primes/misc/lang/golang/primes.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func isPrime(n int) bool {
	ret := n > 1
	for i := 2; i < n; i++ {
		if n%i == 0 {
			ret = false
		}
	}
	return ret
}

func getNumPrimes(start int, len int) int {
	ret := 0
	for k := 0; k < len; k++ {
		if isPrime(start + k) {
			ret++
		}
	}
	return ret
}

func main() {
	start, _ := strconv.Atoi(os.Args[1])
	len, _ := strconv.Atoi(os.Args[2])

	fmt.Println(getNumPrimes(start, len))
}

File: examples/app-primes/misc/lang/r/Primes.R
IsPrime <- function(n) {
    if (n == 1) {
        return(FALSE)
    }
    if (n == 2) {
        return(TRUE)
    }
    for(i in 2:(n-1)){
        if(n %% i == 0){
            return(FALSE)
        }
    }
    return(TRUE)
}

GetNumPrimes <- function(n0,len){
    ret <- 0
    for(i in n0:(n0+len-1)){
        if(IsPrime(i)){
            ret <- ret+1
        }
    }
    return(ret)
}

args <- commandArgs(trailingOnly = TRUE)

start <- as.integer(args[1])
length <- as.integer(args[2])

print (start)
print (length)
numPrimes <- GetNumPrimes(start,length)

print(numPrimes)
